core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -657,6 +657,10 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    *
    * Load data from a flat binary file, assuming the length of each record is constant.
    *
+   * '''Note:''' Normally getBytes returns an array padded with extra values,
```

Comment:
This is referring to the `BytesWritable` values: their `getBytes` methods are technically allowed to return padded byte arrays, so in general callers should take this into account, but in this case we are guaranteeing that `BytesWritable.getBytes` will return a non-padded array.

Actually, this raises a good point: the fact that we're returning un-padded arrays is an implementation detail, so maybe we don't want to make it part of our API contract (unless it's explicitly part of the FixedLengthBinaryInputFormat contract).